rollup/internal/controller/watcher/bundle_proposer.go

```diff
@@ -153,6 +154,11 @@ func (p *BundleProposer) proposeBundle() error {
 
 	hardforkName := encoding.GetHardforkName(p.chainCfg, firstChunk.StartBlockNumber, firstChunk.StartBlockTime)
 	codecVersion := encoding.CodecVersion(batches[0].CodecVersion)
+
+	if codecVersion < encoding.CodecV4 {
```

Comment:
This check is to prevent a misconfiguration via the config file? I guess it does the job but it's not easily maintainable and when having a new version we might forget to set the value correctly. 

imo a better alternative than hard coding would be to make it a mandatory flag when running the relayer. This way we can adjust it dynamically

Reply:
added in [a4e06a5](https://github.com/scroll-tech/scroll/pull/1561/commits/a4e06a539c9916e920ab8b447a30f80520ba8df3).